compiler/rustc_resolve/src/late.rs

```diff
@@ -1578,7 +1576,7 @@ impl<'a, 'ast, 'ra: 'ast, 'tcx> LateResolutionVisitor<'a, 'ast, 'ra, 'tcx> {
                         // Allow all following defaults to refer to this const parameter.
                         forward_const_ban_rib
                             .bindings
-                            .remove(&Ident::with_dummy_span(param.ident.name));
+                            .shift_remove(&Ident::with_dummy_span(param.ident.name));
```

Comment:
The `diesel` regression is due to removals on `bindings`.

Reply:
I reverted it and tried to sort it: https://github.com/rust-lang/rust/pull/131213/commits/818ee0ad11fc56e7e185b5671db7ac2e97dae99c